Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Undo/PrefabUndoUpdateLink.cpp

```diff
@@ -78,15 +89,26 @@ namespace AzToolsFramework
 
             // Remove the link ids if present in the DOMs. We don't want any overrides to be created on top of linkIds because
             // linkIds are not persistent and will be created dynamically when prefabs are loaded into the editor.
-            instanceDom.RemoveMember(PrefabDomUtils::LinkIdName);
-            sourceDom.RemoveMember(PrefabDomUtils::LinkIdName);
+            if (instanceDom.HasMember(PrefabDomUtils::LinkIdName))
+            {
+                AZ_Warning("Prefab", false, "Linked instance DOM from link with id '%llu' shouldn't contains key 'LinkId'",
+                    static_cast<AZ::u64>(linkId));
+                instanceDom.RemoveMember(PrefabDomUtils::LinkIdName);
+            }
+            if (sourceDom.HasMember(PrefabDomUtils::LinkIdName))
+            {
+                AZ_Warning("Prefab", false, "Source template from link with id '%llu' shouldn't contains key 'LinkId'",
+                    static_cast<AZ::u64>(linkId));
+                sourceDom.RemoveMember(PrefabDomUtils::LinkIdName);
+            }
 
             //Diff the instance against the source template.
             PrefabDom linkPatch;
             m_instanceToTemplateInterface->GeneratePatch(linkPatch, sourceDom, instanceDom);
 
             // Create a copy of linkPatch by providing the allocator of m_redoPatch so that the patch doesn't become invalid when
             // the patch goes out of scope in this function.
+            // Todo: Implement a better way to create m_redoPatch instead of copying m_undoPatch to m_redoPatch.
```

Comment:
nit: Remove the TODO as it is not good practice. Add a comment saying that we are copying from undoPatch just to get the 'source' information copied.  Create a github issue instead to recommend a better way.